policyengine/impact/population/charts.py

```diff
@@ -31,38 +31,93 @@ def decile_chart(
     baseline_household_equiv_income = baseline.calc(
         config.equiv_household_net_income_variable
     )
+    reform_household_net_income = reformed.calc(
+        config.household_net_income_variable
+    )
     household_gain = (
-        reformed.calc(config.household_net_income_variable)
-        - baseline_household_net_income
+        reform_household_net_income - baseline_household_net_income
     )
     household_size = baseline.calc("people", map_to="household")
     # Group households in decile such that each decile has the same
     # number of people
     baseline_household_equiv_income.weights *= household_size
     household_decile = baseline_household_equiv_income.decile_rank()
+    agg_gain_by_decile = household_gain.groupby(household_decile).sum()
```

Comment:
On second thought, we could simplify this and avoid the `pd.Series(...).reset_index(drop=True)` statements by using DataFrames more. Something like this (though `mdf.MicroDataFrame` might simplify):
```python
# Create household DataFrame to sum.
w = baseline.weights
household = pd.DataFrame(dict(
    decile = baseline_household_equiv_income.decile_rank(),
    baseline_agg_income = baseline.calc(
        config.household_net_income_variable
    ) * w,
    reform_agg_income = reformed.calc(
        config.household_net_income_variable
    ) * w,
    households = w,
))
# Sum household dataframe by decile.
decile = household.groupby(decile).sum().reset_index()
# Calculate relevant ratios.
decile["baseline_income_per_hh"] = decile.baseline_agg_income / decile.households
# ...
# Create labels.
decile["verb"] = np.where(
        decile.mean_gain > 0,
        "rise",
        np.where(mean_gain < 0, "fall", "remain"),
    )
# ...
```
Then you can just pass the `decile` DataFrame to the chart at the end.

Reply:
I'll file an issue to refactor separately.